### ISSUE TITLE
[BUG] GM.xmlHttpRequest unable to fetch files from WSL

### ISSUE THREAD

**Issue by divinity76** [violentmonkey/violentmonkey#2208]
While fetching files from the C drive like
```js
        GM.xmlHttpRequest({
            method: "GET",
            url: "file:///C:/test.txt",
            onload: function (response) {
                let raw = response.responseText;
                console.log("raw", raw);
            },
            onerror: function (response) {
                console.log("onerror", response);
            },
        });
```
works just fine, trying to fetch a file from WSL fails:
```js
        GM.xmlHttpRequest({
            method: "GET",
            url: "file://wsl.localhost/Ubuntu-24.04/test.txt",
            onload: function (response) {
                let raw = response.responseText;
                console.log("raw", raw);
            },
            onerror: function (response) {
                console.log("onerror", response);
            },
        });
```

### Sequence of actions: 
1. have WSL installed, and inside WSL have a system named "Ubuntu-24.04", and inside that system's root dir, have a file named "test.txt"
2. try to fetch test.txt from WSL via GM.xmlHttpRequest like this:
```js
        GM.xmlHttpRequest({
            method: "GET",
            url: "file://wsl.localhost/Ubuntu-24.04/test.txt",
            onload: function (response) {
                let raw = response.responseText;
                console.log("raw", raw);
            },
            onerror: function (response) {
                console.log("onerror", response);
            },
        });
```
3. the fetch fails. The URL gets translated to `finalUrl: 'http://wsl.localhost/Ubuntu-24.04/test.txt'` which is complete bogus, it should have been `finalUrl: 'file://wsl.localhost/Ubuntu-24.04/test.txt',

### Problem: 
it tries to fetch `http://wsl.localhost/Ubuntu-24.04/test.txt`
### Expected result:  
it should have fetched `file://wsl.localhost/Ubuntu-24.04/test.txt`
#### Devtools console contents:
(probably irrelevant)
#### Environment:
- OS: Windows 10 x64 build 19045.5011
- Browser: Google Chrome Version 129.0.6668.90 (Official Build) (64-bit)
- Violentmonkey Version: 2.26.0

**Comment by divinity76** [comment]
my best guess is that  isLocalUrlRe is failing in https://github.com/violentmonkey/violentmonkey/blob/2dd34c3f729766ed49de190b6876fc43b3dfa84f/src/common/util.js#L257 

but I don't have a good way to test it

**Comment by tophf** [comment]
It can't fail because it matches `file:` at once. I guess it's a bug in the browser.

**Comment by tophf** [comment]
I don't have WSL at the moment, so you can test it by running fetch() in devtools console of Violentmonkey's tab e.g. in dashboard or editor. If it fails too, it's a bug in the browser. A possible workaround is to use XMLHttpRequest, so if it works for you, we'll use it for such urls.

**Comment by divinity76** [comment]
> I don't have WSL at the moment, so you can test it by running fetch() in devtools console of Violentmonkey's tab e.g. in dashboard or editor. If it fails too, it's a bug in the browser. A possible workaround is to use XMLHttpRequest, so if it works for you, we'll use it for such urls.

Seems like a bug in the browser indeed: the url is correct in `sendCmd('HttpRequest', msg);` here
https://github.com/violentmonkey/violentmonkey/blob/master/src/injected/content/requests.js#L72
Then alll of a sudden, the url is wrong in the httpRequested callback here https://github.com/violentmonkey/violentmonkey/blob/master/src/injected/content/requests.js#L94

the stuff that happens inside sendCmd is chrome internals right? so the url gets corrupted inside chrome internals..

yet when i try a normal XMLHttpRequest in violentmonkey tab as you suggested, it works there:
![image](https://github.com/user-attachments/assets/7d21c3eb-8f07-477e-a007-a9cb07bac0d3)

🤷‍♂️ fetch works too

**Comment by tophf** [comment]
I see. The problem is caused by the built-in browser `URL` API, which is used by our getFullUrl function. In case of a normal file:/// URL it ignores our code in `obj.protocol = 'http:';` and keeps the original `file:` protocol. In case of a file://host/ URL it sets the protocol. Looks like a bug in the browser indeed.

**Comment by tophf** [comment]
* `Object.assign(new URL('file://host/foo.txt'), {protocol: 'http:'}).href` --> http://host/foo.txt
* `Object.assign(new URL('file:///foo.txt'), {protocol: 'http:'}).href` --> file:///foo.txt

**Comment by tophf** [comment]
Should be fixed in 8081d778. You can test it as an unpacked extension: [zip](https://github.com/violentmonkey/violentmonkey/actions/runs/11336369615/artifacts/2055466799).

**Comment by divinity76** [comment]
Good job! built with commit e7b4012951dbbdc0b22225845cd685c38d327873 and it works! problem solved :)
